x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/execution_log_table/execution_log_table.tsx

```diff
@@ -0,0 +1,146 @@
+  const [start, setStart] = useState('now-30m');
+  const [end, setEnd] = useState('now');
+  const [filters, setFilters] = useState<string>('');
+
+  const {
+    data: events,
+    isFetching,
+    refetch,
+  } = useRuleExecutionEvents({ ruleId, start, end, filters });
+  const items = events?.events ?? [];
+  const maxEvents = events?.maxEvents ?? 0;
+
+  const onTimeChangeCallback = useCallback(
+    (props: OnTimeChangeProps) => {
+      const recentlyUsedRange = recentlyUsedRanges.filter((range) => {
+        const isDuplicate = range.start === props.start && range.end === props.end;
+        return !isDuplicate;
+      });
+      recentlyUsedRange.unshift({ start: props.start, end: props.end });
+      setStart(props.start);
+      setEnd(props.end);
+      setRecentlyUsedRanges(
+        recentlyUsedRange.length > 10 ? recentlyUsedRange.slice(0, 9) : recentlyUsedRange
+      );
+    },
+    [recentlyUsedRanges]
+  );
+
+  const onRefreshChangeCallback = useCallback((props: OnRefreshChangeProps) => {
+    setIsPaused(props.isPaused);
+    setRefreshInterval(props.refreshInterval);
+  }, []);
+
+  const onRefreshCallback = useCallback(
+    (props: OnRefreshProps) => {
+      refetch();
+    },
+    [refetch]
+  );
+
+  const onSearchCallback = useCallback(
+    ({ queryText }: Parameters<NonNullable<EuiSearchBarProps['onChange']>>[0]) => {
+      setFilters(queryText);
+    },
+    []
+  );
+
+  return (
+    <>
+      <EuiFlexGroup gutterSize="s">
+        <EuiFlexItem style={{ minWidth: '50%' }}>
+          <ExecutionLogSearchBar onSearch={onSearchCallback} />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <div>
+            <EuiSuperDatePicker
+              start={start}
+              end={end}
+              onTimeChange={onTimeChangeCallback}
+              onRefresh={onRefreshCallback}
+              isPaused={isPaused}
+              isLoading={isFetching}
+              refreshInterval={refreshInterval}
+              onRefreshChange={onRefreshChangeCallback}
+              recentlyUsedRanges={recentlyUsedRanges}
+              width="auto"
+            />
+          </div>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+      <EuiSpacer size="s" />
+      <UtilityBar>
+        <UtilityBarSection>
+          <UtilityBarGroup>
+            <UtilityBarText dataTestSubj="executionsShowing">
+              {i18n.SHOWING_EXECUTIONS(items.length)}
+            </UtilityBarText>
+          </UtilityBarGroup>
+          {maxEvents > MAX_EXECUTION_EVENTS_DISPLAYED && (
+            <UtilityBarGroup>
+              <UtilityBarText dataTestSubj="exceptionsShowing">
+                <EuiTextColor color="danger">
+                  {i18n.RULE_EXECUTION_LOG_SEARCH_LIMIT_EXCEEDED(maxEvents)}
+                </EuiTextColor>
+              </UtilityBarText>
+            </UtilityBarGroup>
+          )}
+        </UtilityBarSection>
+      </UtilityBar>
+      <EuiInMemoryTable
```

Comment:
When the auto-refresh option is on, pagination resets on every update, making the table pretty hard to use. It looks like a EUI issue, but I'm not sure.

Reply:
Yeah, I had `refetchOnWindowFocus:false` configured during dev but ended up removing it. It'd definitely be problematic if you're a page or two deep or sorted on a specific field and click away and back and your view resets. Let me see if I can work around this w/ the EUI component, otherwise I'll disable the auto-refresh and we can just rely on filter changes or manual updates via the refresh button.

Reply:
We also might implement pagination on our side without relying on `EuiInMemoryTable`. That's what I did for the rule monitoring table when I stumbled upon similar issues.